examples/native/coin/coin_contract.py

```diff
@@ -1,84 +1,123 @@
+    allowances = nc.Dict(nc.Dict('uint256'))
+
+    def init(ctx, _supply='uint256', returns=STATUS):
+        log.DEV('In Coin.init')
         if isaddress(ctx.owner):
             return FORBIDDEN
         ctx.owner = ctx.tx_origin
-        ctx.holdings[ctx.tx_origin] = 1000000
+        ctx.accounts[ctx.tx_origin] = _supply
+        ctx.supply = _supply
         return OK
 
-    @nc.constant
-    def creator(ctx, returns='address'):
-        return ctx.owner
+    # Standardized Contract API:
+    #   function transfer(address _to, uint256 _value) returns (bool _success)
+    def transfer(ctx, _to='address', _value='uint256', returns=STATUS):
+        log.DEV('In Coin.transfer')
+        if ctx.accounts[ctx.msg_sender] >= _value:
+            ctx.accounts[ctx.msg_sender] -= _value
+            ctx.accounts[_to] += _value
+            ctx.Transfer(ctx.msg_sender, _to, _value)
+            return OK
+        else:
+            return INSUFFICIENTFUNDS
 
-    def sendCoin(ctx, _value='uint32', _to='address', returns=STATUS):
-        if ctx.holdings[ctx.msg_sender] >= _value:
-            ctx.holdings[ctx.msg_sender] -= _value
-            ctx.holdings[_to] += _value
-            ctx.CoinSent(ctx.msg_sender, _value, _to)
+    # Standardized Contract API:
+    #   function transferFrom(address _from, address _to, uint256 _value) returns (bool success)
+    def transferFrom(ctx, _from='address', _to='address', _value='uint256', returns=STATUS):
+        auth = ctx.allowances[_from][ctx.msg_sender]
+        if ctx.accounts[_from] >= _value and auth >= _value:
+            ctx.allowances[_from][ctx.msg_sender] -= _value
+            ctx.accounts[_from] -= _value
+            ctx.accounts[_to] += _value
+            ctx.Transfer(_from, _to, _value)
             return OK
         else:
             return INSUFFICIENTFUNDS
 
+    # Standardized Contract API:
+    #   function totalSupply() constant returns (uint256 supply)
     @nc.constant
-    def coinBalance(ctx, returns='uint32'):
-        return ctx.holdings[ctx.msg_sender]
+    def totalSupply(ctx, returns='uint256'):
+        return ctx.supply
 
+    # Standardized Contract API:
+    #   function balanceOf(address _address) constant returns (uint256 balance)
     @nc.constant
-    def coinBalanceOf(ctx, _holder='address', returns='uint32'):
-        return ctx.holdings[_holder]
+    def balanceOf(ctx, _address='address', returns='uint256'):
+        return ctx.accounts[_address]
+
+    # Standardized Contract API:
+    #   function approve(address _spender, uint256 _value) returns (bool success)
+    def approve(ctx, _spender='address', _value='uint256', returns=STATUS):
+        # approved[msg.sender].withdrawers[_addr]
+        ctx.allowances[ctx.msg_sender][_spender] += _value
+        ctx.Approval(ctx.msg_sender, _spender, _value)
+        return OK
 
-    @nc.constant
-    def numHolders(ctx, returns='uint32'):
-        return len(ctx.holdings)
+    # Standardized Contract API:
```

Comment:
Don't keep dead code. We're using git after all. If you **really** want to include this function as a stub for completeness sake, than raise a https://docs.python.org/2/library/exceptions.html#exceptions.NotImplementedError